kdevnel/ynab-absa-csv-convert-js
# Proces the description and convert to modules

closes #6, closes #11 

## Description
This PR adds the underlying processing for description items and memos. Some description items are not very useful in the current format and so need to be modified and split so that details of who was paid stay in the description column while the other info is moved to the memo column.

### Changelog
* Added description processing
* Split the application into modules
* Converted array keys to lowercase
* Surround memo and description in double quotes

## Testing

* Download the codebase to your computer
* Open index.html using a local server to prevent CORS issues
* Use the /testing/input-sample.csv file in the file uploader
* You should see an array of objects logged to the console and a table containing the formatted CSV data on the page
  * Each object should contain a date, description, memo, amount and balance field.
  * The description and memo fields contain data split from the input CSV description field
  * The description and memo fields should always be surrounded by double quotes, including empty values.
